Make the Kubernetes cluster domain configurable (#6957)
woodpecker-ci/woodpecker: docs/docs/30-administration/10-configuration/11-backends/20-kubernetes.md

With this feature enabled, Woodpecker creates separate Kubernetes namespaces for each organization using the format `{WOODPECKER_BACKEND_K8S_NAMESPACE}-{organization-id}`. Namespaces are created automatically when needed, but they are not automatically deleted when organizations are removed from Woodpecker.

---

### BACKEND_K8S_CLUSTER_DOMAIN

- Name: `WOODPECKER_BACKEND_K8S_CLUSTER_DOMAIN`
- Default: `cluster.local`

The DNS domain of the Kubernetes cluster. It is used to build the DNS search entry that lets a step reach a service by its hostname.

> Set it if your cluster was installed with a custom cluster domain (`--cluster-domain` with kubeadm, `dns_domain` with kubespray). Leaving the default on such a cluster makes services unreachable by hostname.

Reviewer: ```suggestion
Set this value if your cluster was installed with a custom cluster domain.
```
